docs/en/observability/monitor-aws.asciidoc

```diff
@@ -0,0 +1,596 @@
+access logs are useful for many applications. For example, access log
+information can be useful in security and access audits. It can also help you
+learn about your customer base and understand your Amazon S3 bill.
+
+Let's enable the {aws} module in {metricbeat}.
+
+[source,bash]
+----
+./filebeat modules enable aws
+----
+
+Edit the `modules.d/aws.yml` file with the following configurations.
+
+[source,yml]
+----
+- module: aws
+  cloudtrail:
+    enabled: false <1>
+  cloudwatch:
+    enabled: false <1>
+  ec2:
+    enabled: true <2>
+    var.credential_profile_name: fb-aws <3>
+    var.queue_url: https://sqs.eu-central-1.amazonaws.com/836370109380/howtoguide-tutorial <4>
+  elb:
+    enabled: false <1>
+  s3access:
+    enabled: false <1>
+  vpcflow:
+    enabled: false <1>
+----
+<1> Disables the filesets.
+<2> Enables the `ec2` fileset.
+<3> This is the aws profile define following the https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-files.html[AWS standard].
+<4> Add the URL to the queue containing notifications around the bucket containing the EC2 logs
+
+You can now upload your logs to the S3 bucket. If you are using cloudwatch,
+make sure to edit the policy of your bucket like shown in the https://docs.aws.amazon.com/AmazonCloudWatch/latest/logs/S3ExportTasksConsole.html[step 3 of the {aws} user guide]
+it will help you avoiding permissions issues.
+
+Start {filebeat} to collect the logs.
+
+[source,bash]
+----
+./filebeat -e
+----
+
+Let's now configure another fileset and we are going to use the `s3access`
+one. The goal here is to be able to monitor how people access the bucket we
+created. This is what we achieved:
+
+image::one-bucket-archi.png[Current Architecture]
+
+Now what we need to do is to create another bucket and another queue. In order
+to create the following architecture:
+
+image::two-buckets-archi.png[Architecture with Access Logging Enabled]
+
+Make sure the event notifications on the bucket you created is enabled and is
+sending notifications to the new queue you also just created. Now you need to go
+back to the first bucket you created, go to the properties, go to **Server access
+logging**, specify that you want to ship the access logs to the bucket you last
+created.
+
+image::Server-Access-Logging.png[Enabling Server Access Logging]
+
+Copy the URL of the queue you created. Edit the `modules.d/aws.yml`file with
+the following configurations.
+
+[source,yml]
+----
+- module: aws
+  cloudtrail:
+    enabled: false <1>
+  cloudwatch:
+    enabled: false <1>
+  ec2:
+    enabled: true <2>
+    var.credential_profile_name: fb-aws <3>
+    var.queue_url: https://sqs.eu-central-1.amazonaws.com/836370109380/howtoguide-tutorial <4>
+  elb:
+    enabled: false <1>
+  s3access:
+    enabled: true <5>
+    var.credential_profile_name: fb-aws <3>
+    var.queue_url: https://sqs.eu-central-1.amazonaws.com/836370109380/access-log <5>
+  vpcflow:
+    enabled: false <1>
+----
+<1> Disables the filesets.
```

Comment:
Similar thought here in regards to removing the <1>s

Reply:
I removed the ones